### ISSUE TITLE
Persist Deposit Merkle tree

### ISSUE THREAD

**Issue by pawanjay176** [sigp/lighthouse#776]
## Description

The deposit tree is rebuilt from the persisted tree leaves every time for use in the Eth1 caches (once #760 is merged). Rebuilding the deposit tree from scratch for deposit sizes > 1 million takes a few seconds. Encoding the merkle tree as bytes is not straightforward as its a recursive data structure.

## Steps to resolve

Figure out a way to encode/decode the deposit tree and save/load the deposit tree on shutdown/startup.

Blocked on #760

**Comment by michaelsproul** [comment]
Relevant: the `MerkleTree` struct could be deleted and replaced by a `TreeHashCache`, which _is_ easily serializable.

**Comment by michaelsproul** [comment]
This seems like it's probably obsoleted by the deposit snapshots? cc @ethDreamer

**Comment by jimmygchen** [comment]
I believe this is no longer relevant, closing now
